src/CookiePlugin.php

```diff
@@ -109,11 +109,12 @@ private function createCookie(RequestInterface $request, $setCookie)
 
             switch (strtolower($key)) {
                 case 'expires':
-                    $expires = \DateTime::createFromFormat(DATE_COOKIE, $value);
+                    $expires = \DateTime::createFromFormat(DATE_COOKIE, $value)->getTimestamp()
```

Comment:
Well, it looked like the original implementation tried to derive the expiration date from the max-age. But was actually passed as max-age param to the cookie object (original bug). My first commit tried to solve it the other way around, by deriving the max-age from the expiration header. My latest commit passes expiration and max-age to the cookie object as 8th and 3rd constructor arg.